pkg/deployments/src/task.ts

```diff
@@ -286,14 +286,39 @@ export default class Task {
   }
 
   save(output: RawOutput): void {
```

Comment:
I'd break this up into two separate private functions which we select between using the taskmode.

```
 save(output: RawOutput): void {
   if (this.mode === TaskMode.LIVE) {
     this._save(output);
   } else if (this.mode === TaskMode.CHECK) {
     this._smortCheck(output);
   }
```

where `_save()` is the previous function and `_smortCheck()` is the new code.